src/ObjLoading/Game/T6/CustomMap/ProjectCreator.cpp

```diff
@@ -0,0 +1,395 @@
+                    ufbx_vec4 diffuse = mesh->materials.data[i]->fbx.diffuse_color.value_vec4;
+                    vertex->color[0] = static_cast<float>(diffuse.x * factor);
+                    vertex->color[1] = static_cast<float>(diffuse.y * factor);
+                    vertex->color[2] = static_cast<float>(diffuse.z * factor);
+                    vertex->color[3] = static_cast<float>(diffuse.w * factor);
+                    break;
+                }
+
+                default:
+                    _ASSERT(false);
+                }
+                
+
+
+                // 1.0f - uv.y reason:
+                // https://gamedev.stackexchange.com/questions/92886/fbx-uv-coordinates-is-strange
+                ufbx_vec2 uv = ufbx_get_vertex_vec2(&mesh->vertex_uv, index);
+                vertex->texCoord[0] = (float)(uv.x);
+                vertex->texCoord[1] = (float)(1.0f - uv.y);
+
+                ufbx_vec3 normal = ufbx_get_vertex_vec3(&mesh->vertex_normal, index);
+                vertex->normal.x = static_cast<float>(normal.x);
+                vertex->normal.y = static_cast<float>(normal.y);
+                vertex->normal.z = static_cast<float>(normal.z);
+
+                if (mesh->vertex_tangent.exists)
+                {
+                    ufbx_vec3 tangent = ufbx_get_vertex_vec3(&mesh->vertex_tangent, index);
+                    vertex->tangent.x = static_cast<float>(tangent.x);
+                    vertex->tangent.y = static_cast<float>(tangent.y);
+                    vertex->tangent.z = static_cast<float>(tangent.z);
+                }
+                else
+                {
+                    vertex->tangent.x = 0.0f;
+                    vertex->tangent.y = 0.0f;
+                    vertex->tangent.z = 0.0f;
+                }
+
+                vertex->packedLmapCoord = 0;
+
+                // possibly bitangent, unsure what the sign part means though
+                vertex->binormalSign = 0.0f;
+            }
+        }
+
+        _ASSERT(num_vertices == num_triangles * 3);
+
+        // Generate the index buffer.
+        ufbx_vertex_stream streams[1] = {
+            {vertices, num_vertices, sizeof(customMapVertex)},
+        };
+        size_t num_indices = num_triangles * 3;
+        uint32_t* indices = (uint32_t*)calloc(num_indices, sizeof(uint32_t));
+
+        // This call will deduplicate vertices, modifying the arrays passed in `streams[]`,
+        // indices are written in `indices[]` and the number of unique vertices is returned.
+        num_vertices = ufbx_generate_indices(streams, 1, indices, num_indices, NULL, NULL);
+        _ASSERT(num_vertices != 0);
+
+        vertexVec.insert(vertexVec.end(), &vertices[0], &vertices[num_vertices]);
+
+        size_t currIndexVecSize = indexVec.size();
+        indexVec.resize(indexVec.size() + num_indices);
+        for (size_t m = 0; m < num_indices; m++)
+        {
+            indexVec[currIndexVecSize + m] = (uint16_t)indices[m];
```

Comment:
nit: the code mixes C style casts and C++ style casts a lot. The project generally only uses C++ style casts to be more explicit with what should happen and prevent unintended sideeffects.